### PR TITLE
Update developer installation guide

### PR DESCRIPTION
**Pull request by Saransh-cpp**
- use `--recursive` to clone the submodules too (I get a build error locally if `c-blosc` is not cloned)
- use the generic `HTTPS` URLs instead of the `SSH` ones (not everyone has `SSH` keys)
- specify a command to install `numcodecs` globally (I noticed that running `python setup.py build_ext --inplace` did not install `numcodecs` in my `python` environment, but as it correctly built the package, the tests passed)

TODO:

- [ ] Unit tests and/or doctests in docstrings
- [ ] `tox -e py310` passes locally
- [ ] Docstrings and API docs for any new/modified user-facing classes and functions
- [ ] Changes documented in docs/release.rst
- [ ] `tox -e docs` passes locally
- [ ] GitHub Actions CI passes
- [ ] Test coverage to 100% (Coveralls passes)

### CODE REVIEW

File: docs/contributing.rst [hunk 1]
```diff
@@ -75,9 +75,9 @@ The NumCodecs source code is hosted on GitHub at the following location:
 You will need your own fork to work on the code. Go to the link above and hit
 the "Fork" button. Then clone your fork to your local machine::
 
-    $ git clone git@github.com:your-user-name/numcodecs.git
```
Comment (joshmoore) [inline docs/contributing.rst:78]: This goal of this line (based on the proceeding text) is to checkout the user's own fork and not the main repo.

Comment (Saransh-cpp) [reply docs/contributing.rst:78]: Oh yes, missed that!

Comment (joshmoore) [reply docs/contributing.rst:78]: Thanks. So now my only question is if someone doesn't have SSH locally, will this allow them to push a change?

Comment (Saransh-cpp) [reply docs/contributing.rst:78]: Yes! I personally don't use SSH and it works fine.

Comment (jakirkham) [reply docs/contributing.rst:78]: Personally use `ssh` and know the `git@...` syntax is needed to push a change (`https://...` doesn't work)

Comment (Saransh-cpp) [reply docs/contributing.rst:78]: Updated!

Comment (jakirkham) [reply docs/contributing.rst:78]: Thanks! Though AIUI this doesn't work for you. So maybe we should show both syntaxes? Tried a pass at that below

**Comment by Saransh-cpp** [comment]
Thanks for the suggestion! This renders like this right now -

image

**Comment by jakirkham** [comment]
Thanks Saransh! 🙏

Let's see what Josh thinks 🙂

**Comment by joshmoore** [comment]
> Let's see what Josh thinks

Sounds scary 👻 🕷️ ... ;) Anyway, LGTM. Auto-merged engaged! 🚀

File: docs/contributing.rst [hunk 2]
```diff
@@ -75,9 +75,16 @@ The NumCodecs source code is hosted on GitHub at the following location:
...
+    
+or::
+
+    $ git clone --recursive https://github.com/your-user-name/numcodecs.git  # with ``https``
+
+Then ``cd`` into the clone and add the ``upstream`` remote::
+
     $ cd numcodecs
-    $ git remote add upstream git@github.com:zarr-developers/numcodecs.git
+    $ git remote add upstream git@github.com:zarr-developers/numcodecs.git  # or use ``https``
```
Comment (jakirkham) [inline docs/contributing.rst:87]: Actually would just use `https` here. It prevent pushing upstream when using `ssh`, which (to me at least) is a feature and not a bug :)

```suggestion
    $ git remote add upstream https://github.com/zarr-developers/numcodecs.git
```

**Comment by jakirkham** [comment]
Thanks Josh!

Sorry had missed one thing. Added that above. Otherwise LGTM

**Pull request merged**